src/ngraph/op/fused/lstm_cell.hpp

```diff
@@ -0,0 +1,199 @@
+            ///                               [4*hidden_size, hidden_size].
+            /// \param[in]  H_t               The hidden state tensor at current time step with
+            ///                               shape: [batch_size, hidden_size].
+            /// \param[in]  C_t               The cell state tensor at current time step with
+            ///                               shape: [batch_size, hidden_size].
+            /// \param[in]  hidden_size       The number of hidden units for recurrent cell.
+            /// \param[in]  B                 The bias tensor for input gate with shape: [8*hidden_size].
+            /// \param[in]  P                 The weight tensor for peepholes with shape:
+            ///                               [3*hidde_size] - 3 equals to only iof gates.
+            /// \param[in]  activations       The vector of activation functions used inside
+            ///                               recurrent cell.
+            /// \param[in]  activation_alpha  The vector of alpha parameters for activation
+            ///                               functions in order respective to activation list.
+            /// \param[in]  activation_beta   The vector of beta parameters for activation functions
+            ///                               in order respective to activation list.
+            /// \param[in]  clip              The value defining clipping range [-clip, clip] on
+            ///                               input of activation functions.
+            /// \param[in]  input_forget      Controls coupling input and forget gates.
+            ///
+            LSTMCell(const std::shared_ptr<Node>& X,
+                     const std::shared_ptr<Node>& W,
+                     const std::shared_ptr<Node>& R,
+                     const std::shared_ptr<Node>& H_t,
+                     const std::shared_ptr<Node>& C_t,
+                     std::size_t hidden_size,
+                     const std::shared_ptr<Node>& B,
+                     const std::shared_ptr<Node>& P,
+                     const std::vector<std::string>& activations =
+                         std::vector<std::string>{"sigmoid", "tanh", "tanh"},
+                     const std::vector<float>& activation_alpha = {},
+                     const std::vector<float>& activation_beta = {},
+                     float clip = 0.f,
+                     bool input_forget = false);
+
+            virtual void pre_validate_and_infer_types() override;
+            virtual NodeVector decompose_op() const override;
+            virtual std::shared_ptr<Node>
+                copy_with_new_args(const NodeVector& new_args) const override;
+
+            bool get_input_forget() const { return m_input_forget; }
+        private:
+            ///
+            /// \brief      The input data tensor. Shape: [batch_size, input_size].
+            ///
+            std::shared_ptr<Node> m_X;
```

Comment:
I don't think it will work to store the node pointers here. If we do a rewrite on the graph prior to fused op decomposition (which is allowed), replace_output will not know how to rewire these pointers. You'll need to refer to them according to input index.

Reply:
Same comment for m_W, m_R, m_H_t, m_C_t, m_p_iof, and m_bias.